docs/reference/module-api/module-overview.md

```diff
@@ -483,18 +483,354 @@ module.exports = {
+Valid names for functions returned by `restApiRoutes` include:
+- `getAll`
+- `getOne`
+- `post`
+- `patch`
+- `put`
+- `delete`
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  restApiRoutes(self) {
+    return {
+      // GET /api/v1/product
+      async getAll(req) {
+        const results = [];
+        // ... populate `results` with product data.
+
+        return {
+          results
+        };
+      }
+    };
+  }
+};
+```
+
 #### `extendRestApiRoutes(self)`
+
+Extend the behavior of existing REST API routes in the `extendRestApiRoutes` section. This function must return an object of functions. See [`restApiRoutes`](#restapiroutes-self) for the valid function names.
+
+Each extended REST API route function should accept the original function as `_super` and the `req` request object. They should return data in a similar format to the existing [piece](/reference/api/pieces.md) and [page](/reference/api/pages.md) REST API (e.g., single doc `GET` requests should return a single document object and general `GET` requests should return an object including a `result` array of document objects).
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  extendRestApiRoutes(self) {
+    return {
+      // GET /api/v1/product
+      async getAll(_super, req) {
+        // Get the original function's response (making sure to `await`).
+        const response = await _super(req);
+
+        if (Array.isArray(response.results)) {
+          // Adds a `resultLength` property on the response object.
+          response.resultLength = response.results.length;
+        }
+
+        return response;
+      }
+    };
+  }
+};
+```
+
+### `apiRoutes(self)`
+
+Add custom API routes. The `apiRoutes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  apiRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/cheapest
+        async cheapest(req) {
+          const product = await self.find(req).sort({
+            price: 1
+          }).toObject();
+          if (!product) {
+            // Browser receives a 404 error
+            throw self.apos.error('notfound', 'No products were found.');
+          }
+          // Response is a JSON object with a `product` property
+          return {
+            product
+          };
+        }
+      }
+    };
+  }
+};
+```
+
+#### Naming routes
+
+If route properties do *not* begin with a forward slash, they can be reached via the pattern: `/api/v1/` followed by the module name, a forward slash, and the API route handler's property name, e.g., `/api/v1/product/cheapest` for the `cheapest` route on the `product` module.
```

Comment:
`cheapest` could probably be replaced by something that translates a bit better, since its pretty much slang. Using product....my first thought is `brand/s` or a category like `denim-jacket/s`. Denim and jacket are both fairly universal, while being very American!

Bonus points if we find a way to use infant bodysuit

Reply:
😕 "cheapest" simply means "most cheap." Least expensive.